reactor-core/src/test/java/reactor/core/scheduler/ElasticSchedulerTest.java

```diff
@@ -276,4 +281,64 @@ else if (i == fastCount - 1) {
 			LOGGER.info("{} threads active post shutdown", Thread.activeCount() - otherThreads);
 		}
 	}
+
+	@Test
+	public void doesntRecycleWhileRunningAfterDisposed() throws Exception {
+		Scheduler s = Schedulers.newElastic("test-recycle");
+		((ElasticScheduler)s).evictor.shutdownNow();
+
+		try {
+			AtomicBoolean stop = new AtomicBoolean(false);
+			CountDownLatch started = new CountDownLatch(1);
+			Disposable d = s.schedule(() -> {
+				started.countDown();
+				// simulate uninterruptible computation
+				for (;;) {
+					if (stop.get()) {
+						break;
+					}
+				}
+			});
+			started.await();
```

Comment:
maybe prevent infinite test (in case of future regression) by capping the `await` and checking the returned boolean?

Reply:
@dittos apart from this minor change to the test, this PR should be good to go

```suggestion
			assertThat(started.await(10, TimeUnit.SECONDS)).as("latch timeout").isTrue();
```